spring-integration-core/src/main/java/org/springframework/integration/dsl/RecipientListRouterSpec.java

```diff
@@ -122,7 +124,7 @@ private static boolean isLambda(Object o) {
 	 * @return the router spec.
 	 */
 	public RecipientListRouterSpec recipient(MessageChannel channel) {
-		return recipient(channel, (String) null);
+		return recipient(channel, new ValueExpression<Boolean>(Boolean.TRUE));
```

Comment:
I think would be better to rework the target method as:
```
public RecipientListRouterSpec recipient(MessageChannel channel, Expression expression) {
        if (expression != null) {
              ExpressionEvaluatingSelector selector = new ExpressionEvaluatingSelector(expression);
              this.target.addRecipient(channel, selector);
              this.componentsToRegister.add(selector);
        }
        else {
              this.target.addRecipient(channel);
        }
	return _this();
}
```

to save some memory.

Let me know and I'll fix on merge